fuel-jackson/src/test/kotlin/com/github/kittinunf/fuel/FuelJacksonTest.kt

```diff
@@ -38,25 +40,33 @@ class FuelJacksonTest : MockHttpTestCase() {
 
         Fuel.get(mock.path("user-agent"))
             .responseObject(jacksonDeserializerOf<HttpBinUserAgentModel>()) { _, _, result ->
-                assertThat(result.component1(), instanceOf(HttpBinUserAgentModel::class.java))
-                assertThat(result.component1()?.userAgent, not(""))
-                assertThat(result.component2(), instanceOf(FuelError::class.java))
+                assertThat(result, instanceOf(Result.Success::class.java))
+                with(result as Result.Success) {
+                    assertThat(value, instanceOf(HttpBinUserAgentModel::class.java))
+                    assertThat(value.userAgent, not(""))
+                }
             }
+            .get()
```

Comment:
This ensures that assertions in the asynchronous callback are able to fail the tests.